lib/internal/modules/esm/module_job.js

```diff
@@ -75,6 +75,19 @@ const isCommonJSGlobalLikeNotDefinedError = (errorMessage) =>
 const explainCommonJSGlobalLikeNotDefinedError = (e, url) => {
   if (e?.name === 'ReferenceError' &&
       isCommonJSGlobalLikeNotDefinedError(e.message)) {
+
+    if (e.message.includes('require is not defined')) {
```

Comment:
Checking for `require is not defined` seems like it belongs within `isCommonJSGlobalLikeNotDefinedError`.